src/api/java/com/minecolonies/api/crafting/RecipeStorage.java

```diff
@@ -329,29 +448,36 @@ public boolean fullfillRecipe(final List<IItemHandler> handlers)
      *
      * @param handlers the handlers.
      */
-    private void insertCraftedItems(final List<IItemHandler> handlers)
+    private void insertCraftedItems(final List<IItemHandler> handlers, ItemStack outputStack)
     {
         for (final IItemHandler handler : handlers)
         {
-            if (InventoryUtils.addItemStackToItemHandler(handler, getPrimaryOutput().copy()))
+            if (InventoryUtils.addItemStackToItemHandler(handler, outputStack.copy()))
             {
                 break;
             }
         }
 
         final List<ItemStack> secondaryStacks = new ArrayList<>();
-        for (final ItemStack stack : input)
+        if(!secondaryOutputs.isEmpty())
         {
-            if (stack.getItem() == ModItems.buildTool)
+            secondaryStacks.addAll(secondaryOutputs);
+        }
+        else
+        {
+            for (final ItemStack stack : input)
             {
-                continue;
-            }
+                if (stack.getItem() == ModItems.buildTool)
```

Comment:
You are in the API, no special casing of Minecolonies specific stuff.
Create a new Interface if you want to and smag that on the buildtool then do an instance of.
Or create a behaviour registry.

Regardless either refactor this class so that it is not part of the API or refactor it so that it does not do any special casing.

Reply:
Ok, the buildTool exception is pre-existing. We make this exception in 3 places in this file.  Changing this item is a Structurize change.